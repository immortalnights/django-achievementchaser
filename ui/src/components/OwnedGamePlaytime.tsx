import { Typography } from "@mui/material"
import { duration } from "../utilities"

const OwnedGamePlaytime = ({ playtimeForever }: PlayerOwnedGame) => {
    const value = playtimeForever
        ? `${duration(playtimeForever).asHours().toFixed(1)} hours`
        : "None"

    return (
        <div>
            <Typography variant="subtitle1" textTransform="uppercase">
                Play Time
            </Typography>
            <div>{value}</div>
        </div>
    )
}

export default OwnedGamePlaytime
