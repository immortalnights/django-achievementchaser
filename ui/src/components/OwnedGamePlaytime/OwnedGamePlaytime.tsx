import { Typography } from "@mui/material"
import { duration } from "@/dayjsUtilities"

const OwnedGamePlaytime = ({ playtimeForever }: GameOwner) => {
    const value = playtimeForever
        ? `${duration(playtimeForever).asHours().toFixed(1)} hours`
        : "None"

    return (
        <div>
            <Typography variant="subtitle1" textTransform="uppercase">
                Play Time
            </Typography>
            <div>{value}</div>
        </div>
    )
}

export default OwnedGamePlaytime
