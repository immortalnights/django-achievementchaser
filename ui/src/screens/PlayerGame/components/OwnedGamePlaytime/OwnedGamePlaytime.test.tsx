import renderer from "react-test-renderer"
import { expect, test } from "vitest"
import OwnedGamePlaytime from "."
import { playedPlayerOwnedGame } from "@test/data"

test("snapshot", () => {
    // eslint-disable-next-line
    const component = renderer.create(
        <OwnedGamePlaytime {...playedPlayerOwnedGame} />
    )
    expect(component.toJSON()).toMatchSnapshot()
})
