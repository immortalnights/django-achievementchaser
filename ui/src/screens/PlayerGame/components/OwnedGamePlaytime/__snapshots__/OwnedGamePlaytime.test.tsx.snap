// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1s1s9f7-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.75;letter-spacing:0.00938em;text-transform:uppercase;}",
      }
    }
    data-emotion="css 1s1s9f7-MuiTypography-root"
  />
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 css-1s1s9f7-MuiTypography-root"
    style={{}}
  >
    Play Time
  </h6>
  <div>
    7.8 hours
  </div>
</div>
`;
